src/lib/schema.js

```diff
@@ -83,6 +83,17 @@ export function getMetricFromKey(key, bucket) {
 }
 
 /**
+* Returns metric from a state key
+* @param {string} key - schema key
+* @param {string} bucket - bucket name
+* @return {string} metric - S3 metric
+*/
+export function getMetricFromStateKey(key, bucket) {
+    // s3:buckets:foo:storageUtilized
+    return key.slice(11).replace(`${bucket}:`, '');
```

Comment:
Could return what is after the last colon, otherwise if that does not work, you may use `'s3:buckets'.length` instead of the hard-coded 11 after checking that's how the string starts (unless there's a potential big performance impact doing this).